stdlib/3/tkinter/font.pyi

```diff
@@ -0,0 +1,109 @@
+import tkinter
+from typing import Any, List, Optional, Tuple, Union, overload
+from typing_extensions import Literal, TypedDict
+
+NORMAL: Literal["normal"]
+ROMAN: Literal["roman"]
+BOLD: Literal["bold"]
+ITALIC: Literal["italic"]
+
+def nametofont(name: str) -> Font: ...
+
+# See 'FONT DESCRIPTIONS' in font man page. This uses str because Literal
+# inside Tuple doesn't work.
+_FontDescription = Union[
+    str, Font, Tuple[str, int], Tuple[str, int, Tuple[str, ...]], Tuple[str, int, List[str]],
+]
+
+class _FontDict(TypedDict):
+    family: str
+    size: int
+    weight: Literal["normal", "bold"]
+    slant: Literal["roman", "italic"]
+    underline: Literal[0, 1]
+    overstrike: Literal[0, 1]
+
+class _MetricsDict(TypedDict):
+    ascent: int
+    descent: int
+    linespace: int
+    fixed: Literal[0, 1]
+
+class Font:
+    name: str
+    delete_font: bool
+    def __init__(
+        self,
+        # In tkinter, 'root' refers to tkinter.Tk by convention, but the code
+        # actually works with any tkinter widget so we use tkinter.Misc.
+        root: Optional[tkinter.Misc] = ...,
+        font: Optional[_FontDescription] = ...,
+        name: Optional[str] = ...,
+        exists: bool = ...,
+        *,
+        family: str = ...,
+        size: int = ...,
+        weight: Literal["normal", "bold"] = ...,
+        slant: Literal["roman", "italic"] = ...,
+        underline: bool = ...,
+        overstrike: bool = ...,
+    ) -> None: ...
+    @overload
+    def cget(self, option: Literal["family"]) -> str: ...
+    @overload
+    def cget(self, option: Literal["size"]) -> int: ...
+    @overload
+    def cget(self, option: Literal["weight"]) -> Literal["normal", "bold"]: ...
+    @overload
+    def cget(self, option: Literal["slant"]) -> Literal["roman", "italic"]: ...
+    @overload
+    def cget(self, option: Literal["underline", "overstrike"]) -> Literal[0, 1]: ...
+    __getitem__ = cget
+    @overload
+    def actual(self, option: Literal["family"], displayof: Optional[tkinter.Misc] = ...) -> str: ...
+    @overload
+    def actual(self, option: Literal["size"], displayof: Optional[tkinter.Misc] = ...) -> int: ...
+    @overload
+    def actual(self, option: Literal["weight"], displayof: Optional[tkinter.Misc] = ...) -> Literal["normal", "bold"]: ...
+    @overload
+    def actual(self, option: Literal["slant"], displayof: Optional[tkinter.Misc] = ...) -> Literal["roman", "italic"]: ...
+    @overload
+    def actual(self, option: Literal["underline", "overstrike"], displayof: Optional[tkinter.Misc] = ...) -> Literal[0, 1]: ...
+    @overload
+    def actual(self, option: None, *, displayof: Optional[tkinter.Misc] = ...) -> _FontDict: ...
+    @overload
+    def actual(self, *, displayof: Optional[tkinter.Misc] = ...) -> _FontDict: ...
```

Comment:
I think technically these last two overloads should be:
```suggestion
    @overload
    def actual(self, option: None = ..., displayof: Optional[tkinter.Misc] = ...) -> _FontDict: ...
```
since e.g, you could pass None, None positionally

Reply:
How would the type checker know that `some_font.actual()` is same as passing `None` for the option? Maybe just remove `*` from the `actual(self, option: None, *, ...)` overload?

Reply:
Because it's the only overload where both parameters have default values.
What you've done also works :-)